Doc: update changelog (#13187)
OpenTTD/OpenTTD: changelog.md

- Fix: Reverse left/right keypress when editing RTL text (#12711, #12710)
- Fix: Crash if interface scale or font size changes with chat box present (#12705)
- Fix: Make progress bars obey language direction (#12704)
- Fix #12681: Abstract filetype not set for network client join savegames (#12701)
- Fix: Implement support for secure state coding on macOS (#12698)
- Fix: Add entitlements needed for plugins on macOS (#12697)
- Fix: Allow resolving house parent scope (town) of unbuilt houses (#12695)
- Fix #12685: Nullptr dereference when checking for equal loaded/loading groups (#12686)
- Fix: Unable to choose a font containing hyphen (#12684)
- Fix: [Timetable] Use days as precision in day mode for accurate timetable syncing (#12683)
- Fix #12550: Files were not saved in the right location when binary and configuration are in the same folder (#12679)
- Fix #12591: Give descriptive error when station construction fails due to wrong layout (#12678)
- Fix #12594: Give descriptive error when company takeover fails due to vehicle limit (#12676)
- Fix: Ensure revision mismatch also considers random debugging status (#12673)

> - Fix: [Admin] #12411 Send Network Welcome Packet to admin port after game creation completed WelcomeAll moved into NetworkOnGameStart (#12672)

Reviewer: ```suggestion
- Fix #12411: [Admin] Send Network Welcome Packet to admin port after game creation completed WelcomeAll moved into NetworkOnGameStart (#12672)
```
Unless that was a deliberate choice?

- Fix #12563: Race condition setting finish flag in WinHTTP (#12658)
- Fix #12648: Ensure all uses of std::filesystem::path use native encoding (#12650)
- Fix #12643: _is_water_region_valid is not cleared/reset in AllocateWaterRegions (#12649)
- Fix: Console command scrolling did not take account of padding and cursor width (#12642)
- Fix: Allow changing size of default OpenTTD font (#12641)
- Fix: Always allow setting company settings, company/president name/face (#12635)
- Fix #12608: SDL keycode to vkey mapping (#12631)
- Fix: For GUI network servers, name the first company the same as any other company (#12629)
- Fix: Server's client is shown incorrectly in some cases (#12629)
- Fix: Properly test for presence of waypoint in NewGRF resolver (#12579)
- Fix #12497: [NewGRF] Increase vehicle motion counter for all parts (#12561)
- Fix: Inconsistent space between console history and current line (#12528)
- Fix: Industries accept/produce no cargo for pre-SLV_78 saves (#12508)
- Fix: Station/industry nearby list checks in CheckCaches (#12505)
- Fix: Build industry window did not take width of count into account (#12476)

> - Fix: Sym->sym accidentally changed to sym prevented keys working with SDL (#12454)

Reviewer: ```suggestion
```

## 15.x

### 15.0-beta1 (2024-12-24)

> - Feature: Allow manually placing town buildings in scenario editor (#12661)

Reviewer 2: ```suggestion
- Feature: Manually place town buildings in scenario editor (#12661, #12741)
```

- Feature: Ctrl-click to toggle favourites in build-pickers (#12595)
- Feature: New filter to show only used types in build-pickers (#12595)

> - Feature: Add "All" filter to build-picker show types from all classes (#12595)

Reviewer 2: ```suggestion
- Feature: Filter new picker window by all classes, by used types, or by favourites (#12595)
- Feature: New picker window for rail and road stations, rail and road waypoints, and objects  (#12595)
```

> - Feature: Road waypoint functionality (#12572)

Reviewer 2: ```suggestion
- Feature: Road waypoints (#12572)
```

> - Feature: [NewGRF] Allow fixed layout up to 256 tiles per NewGRF rail station (#12554)

Reviewer 2: Should this be a `Feature`, or just an `Add`, since it's only relevant for NewGRF authors?

- Feature: Allow base sounds set to be changed mid-game (#12399)
- Feature: Authorize specific clients to join network company without password (#12337)
- Feature: Authorized key authentication for rcon (#12326, #12329)

> - Feature: Authenticate to the server without sending the password (#12323, #11878)

Reviewer: These three are kinda about the same topic, so maybe combine them? Not sure though, sounds quite technical either way 🤷 
```suggestion
- Feature: Password-less network authentication and authorization (#11878, #12323, #12326, #12329, #12337)
```

Author: That covers only the first two, as 12323 and 11878 are not about password-less auth. They still use a password, it's just that the password is not sent over the wire in any form, but rather the password and a new random token for each call is used to generate the encryption keys and with that connection is encrypted. If you don't know the password, then you can't get through decryption and message validation.

> - Feature: Authenticate to the server without sending the password (#12323, #11878)

Reviewer 2: Are these visible to players or are they behind-the-scenes changes? If they are visible to server owners, perhaps a [server] tag would clarify?

Author: > Are these visible to players or are they behind-the-scenes changes? If they are visible to server owners, perhaps a [server] tag would clarify?

With rcon it's questionable whether those users are still players, and the authentication without sending the passwords is invisible to the end users (except some secrets being written).

On the other hand: stepping away from sending passwords over the network is quite a major change, as beforehand a malicious server could store the hashes and reverse engineer your password by brute force. With the new way, there is no hash to work against. So, it's not directly player visible, but a huge step forward in  security.

- Feature: Encrypt the connection between game server and client, and admin (#12300, #11878)

> - Feature: Industry production graph (#10541)

Reviewer 2: ```suggestion
- Feature: Industry production/transported graph (#10541, #13083)
```

- Feature: Import town data from JSON file (#10409)

> - Add: Implement 2D map scrolling under SDL2 (#13167) (#13169)

Reviewer: ```suggestion
- Add: Implement 2D map scrolling under SDL2 (#13169)
```

Or drop #13167 completely as it is an issue, not a PR.

- Add: [NewGRF] Callback for custom refit mask for engines/vehicles (#13090)
- Add: [NewGRF] Vehicle prop that allows refittability based on cargo class intersection (#13090)

> - Add: Add cargo transported to industry production graph (#13083)

Reviewer 2: I suggest listing this with #10541.

```suggestion
```

- Add: Ability to show and filter different datasets on the same graph (#13083)
- Add: Show and toggle follow vehicle state in vehicle view window (#13076)
- Add: Set application name for SDL2 (#13061)
- Add: Support sound effects in Ogg Opus format (#13055)
- Add: Improving town-owned bridges increases company rating (#13036)
- Add: Apply rail/road type conversion when NewGRFs are updated mid-game (#13021)
- Add: Implement missing road type label conversion (#13021)
- Add: [NewGRF] Cargo class bits 13 and 14 now define potable and non-potable (#12979)
- Add: Overlay cargo icon in vehicle/depot list when holding shift+ctrl (#12938)
- Add: [Script] Include number of victims in ScriptEventVehicleCrashed (#12861)

> - Add: Town, industry and vehicle window zoom with mouse wheel (#12810, #12809, #12797)

Reviewer 2: I might count this as a `Feature`.

> - Add: Show count of towns and industries in their directory window captions (#12800)

Reviewer 2: Also a `Feature`, I think.

> - Add: Toggle for showing industry names in small map (#12770)

Reviewer 2: Another `Feature`.

- Add: [Console] Schedule command to execute a script file next in-game month (#12761)

> - Add: Show house information in house placer (#12741)

Reviewer 2: Part of #12661.

```suggestion
```

- Add: Setting to disable warning for old vehicles (#12714)
- Add: Use macOS Game Mode identification (#12700)
- Add: Shade button to waypoint picker (#12682)

> - Add: Command to build an individual house on a specific tile (#12661)

Reviewer 2: Part of #12661, not its own changelog item.

```suggestion
```

- Add: [NewGRF] Extended custom waypoint classes (#12653)
- Add: 'Get Content' buttons next to base set dropdowns in Game Options (#12627)
- Add: [NewGRF] Custom road waypoint support (#12572)
- Add: [NewGRF] Station property 1E, extended station tile flags (#12554)
- Add: [SDL2] Driver parameter 'no_mouse_capture' to ease interactive debugging (#12336)
- Add: Basic autocompletion on tab for console commands (#12163)
- Add: Portuguese Escudo currency (#12108)
- Change: Hide buttons in Found Town window that can't be used (#13182)
- Change: Include new cargo classes in dump cargo types console command (#13174)
- Change: Use floats for tracking 2D scrolling to improve smoothness (#13169)
- Change: Improve news window layouts (#13136)

> - Change: [UI] Improve visual padding of vehicle news items (#13134)

Reviewer: Combine?

- Change: Show company finances column if it has any values in it (#13112)
- Change: Treat recolour sprites as regular sprites in the SpriteCache (#13107)
- Change: Determine industry directory width only on visible rows (#13097)
- Change: Add sound memory usage to framerate window (#13055)
- Change: Invalidate build toolbars when NewGRFs are changed (#13046)
- Change: Don't try to flood buoys or dock tiles (#13013)
- Change: Store water tile flooding state in the map (#13013)
- Change: Exclude parent item from directory sort in FiosGetFileList (#12947)
- Change: [NewGRF] Place all 8 bits of station tile layout in var 40/41 (#12890)
- Change: [NewGRF] Install translation tables into overridden NewGRF (#12879)
- Change: [UI] Use scaled sprite sizes and correct matrix padding for content list (#12875)

> - Change: [UI] Use proper matrix padding for server list test (#12874)
> - Change: [UI] Use interface scaled sizes for server list (#12874)

Reviewer: ```suggestion
- Change: [UI] Improved Network server list icon sizing and text positioning (#12874)
```

- Change: Path signals now show green on junction-less tracks by default (#12857)
- Change: Don't use house construction states in Scenario Editor (#12822)

> - Change: Improve when to stops following vehicle (#12808)

Reviewer 2: ```suggestion
- Change: Changing zoom no longer stops following vehicle (#12808)
```

> - Change: Improve when to stops following vehicle (#12808)

Reviewer: ```suggestion
- Change: Improve when to stop following vehicles (#12808)
```

- Change: Allow bribing local authority when other company has exclusive rights (#12763)
- Change: Position caret on left or right of glyph depending on language direction (#12760)
- Change: Do not automatically connect double depots with track (#12752)
- Change: Call custom house name callback in House Picker (#12741)
- Change: [NewGRF] Treat house max year 0xFFFF as MAX_YEAR (#12740)
- Change: [UI] Remove substitution of missing glyph with '?' glyph (#12736)
- Change: [UI] Use SetMinimalTextLines to set label height correctly (#12734)
- Change: Don't use house construction states in Scenario Editor (#12730)

> - Change: Don't reinitialise font caches when setting initial interface scale (#12722)

Reviewer: Not really player-facing I think.

```suggestion
```

- Change: Ignore min/max years for manual house placer (#12687)

> - Change: Allow sprite resolver to evaluate non-existent houses (#12661)

Reviewer 2: This is an internal change and shouldn't be listed.

```suggestion
```

> - Change: Allow sprite resolver to evaluate non-existent houses (#12661)

Reviewer: Not really player-facing I think.
```suggestion
```

> - Change: Use default NewGRF cargo translation table (#12646)

Reviewer: ```suggestion
- Change: Use default NewGRF cargo translation table if no custom table is provided (#12646)
```

> - Change: Use button aspect ratio for buttons that show vehicle icons (#12636)
> - Change: Don't include midi file processing in dedicated server build (#12632)
> - Change: [Linkgraph] Improve distance scaling algorithm in demand scaler (#12542)
> - Change: Draw group hierarchy tree lines (#12522)
> - Change: Display more useful information in sprite aligner than sprite ID (#12439)
> - Change: Disallow using Action A to load sprites above the baseset unless reserved (#12435)
> - Change: Use aspect ratios for some common widgets and buttons (#12386)

Reviewer: Combining #12636 and #12386.

```suggestion
- Change: Don't include midi file processing in dedicated server build (#12632)
- Change: [Linkgraph] Improve distance scaling algorithm in demand scaler (#12542)
- Change: Draw group hierarchy tree lines (#12522)
- Change: Display more useful information in sprite aligner than sprite ID (#12439)
- Change: Disallow using Action A to load sprites above the baseset unless reserved (#12435)
- Change: Use aspect ratios for some common widgets and buttons (#12386, #12636)
```

- Change: [NewGRF] Increase house type limit from 512 to 4096 (#12288)
- Change: Add dividers in vehicle group action dropdown (#12284)
- Change: Use same audio buffer size (and config parameter) for all sound drivers (#12227)
- Change: Allow rail and road depot overbuilding in current orientation in order to connect to rail or road (#12219)
- Change: Show tile index as decimal number in land info window (#12104)
- Change: Increase object/station/roadstop class limit (#12094)
- Change: Disable the insecure admin login by default; replaced by allow list (#11878)
- Fix: [NewGRF] New engines did not have a default cargo type set (#13146)
- Fix: String parameter not set when determining width of smallmap contour labels (#13145)
- Fix: Incorrect translation table used for older NewGRFs (#13131)
- Fix #13022: Ensure minimum size of scrollbar slider (#13119)
- Fix #13110: [Script] Convert table keys to string when generating JSON (#13113)
- Fix: Error message not set when unable load an old save (#13106)
- Fix: Network clients incorrectly truncated all strings to NETWORK_COMPANY_NAME_LENGTH (#13102)
- Fix: Don't allow right-click to close world generation progress window (#13084)

> - Fix: VF_LOADING_FINISHED and VF_STOP_LOADING flags tested on wrong vehicle part (#13063)

Reviewer: Not sure this is player-facing, but there's no issue to reference, so dunno

- Fix: X-axis direction of industry production graph (#13062)
- Fix #13053: Payment transfers incorrect for non-passenger cargos (#13054)
- Fix: Build with icu >= 76 where icu-i18n and icu-uc become separated (#13048)
- Fix: Terraform of road waypoint tiles (#13040)
- Fix: Invalid data used for height map curves after first run (#13039)
- Fix: Skip control codes when sorting strings (#13035)
- Fix: SkipGarbage() skipped all multi-byte utf-8 characters (#13032)
- Fix: Don't invalidate water regions on the other side of the map (#13012)
- Fix: ScriptTile::PlantTreeRectangle does not check that the end tile is in bounds (#13004)
- Fix #12973: Don't exclude high score after using sandbox (#12999)
- Fix #12976: Incorrect widget rect scrolling for RTL languages (#12978)
- Fix: House animation random data flag ignored (#12963)
- Fix: Crash when exiting game with end game window open (#12962)

> - Fix: TC_FORCED no longer working (#12961)

Reviewer: ```suggestion
```

- Fix: Crash when pressing arrow keys in bootstrap mode (#12948)
- Fix #12945: Relative paths passed to -c were not made absolute (#12946)
- Fix: Erroneous error message (#12944)
- Fix #12940: Use specific error when overbuilding station on signals (#12943)
- Fix: Crash in GetGameStringPtr when there are no GS strings (#12933)
- Fix #12917: Write to negative array index for some string patterns (#12932)
- Fix: Road stop availability callback (#12931)
- Fix: Road stop variable 6B crashing for road stop preview in GUI (#12930)
- Fix: Incorrect truncation in string setting validation (#12924)
- Fix #12914: Fix use of invalidated pointer in viewport drawer (#12918)
- Fix: Picker window 'used' filter for rail waypoints (#12911)
- Fix: Train curve detection did not take shortened parts into account (#12910)
- Fix #12901: Savegame format of company allow lists (#12908)
- Fix: DepotWindow::GetVehicleFromDepotWndPt not refreshing vehicle list (#12907)
- Fix: DupSprite did not copy the control_flags field (#12906)
- Fix: Excessively fast ships on aqueducts fail to move at correct speed (#12895)
- Fix: Train unit ID allocation when moving engine to new train at train limit (#12887)

> - Fix: Remove the invalid UTF-8 character (U+200B) from the traditional chinese translation (#12882)

Reviewer: ```suggestion
```

- Fix #12655: Inconsistent state in client list and potential crash after client leaves (#12660)
- Fix: Use widget resize step instead of window resize step (#12659)
- Fix #12563: Race condition setting finish flag in WinHTTP (#12658)
- Fix #12648: Ensure all uses of std::filesystem::path use native encoding (#12650)
- Fix #12643: _is_water_region_valid is not cleared/reset in AllocateWaterRegions (#12649)
- Fix: Console command scrolling did not take account of padding and cursor width (#12642)
- Fix: Allow changing size of default OpenTTD font (#12641)
- Fix: Always allow setting company settings, company/president name/face (#12635)
- Fix #12608: SDL keycode to vkey mapping (#12631)
- Fix: For GUI network servers, name the first company the same as any other company (#12629)
- Fix: Server's client is shown incorrectly in some cases (#12629)
- Fix: Properly test for presence of waypoint in NewGRF resolver (#12579)
- Fix #12497: [NewGRF] Increase vehicle motion counter for all parts (#12561)
- Fix: Inconsistent space between console history and current line (#12528)
- Fix: Industries accept/produce no cargo for pre-SLV_78 saves (#12508)

> - Fix: Use modern comparisons instead of memcmp in cache check (#12507)

Reviewer 2: This sounds more like a codechange than a fix.
```suggestion
```
